packages/jest-runtime/src/index.ts

```diff
@@ -1369,7 +1369,7 @@ class Runtime {
 
     Object.defineProperty(moduleRequire, 'main', {
       enumerable: true,
-      value: this._testPath,
+      value: this._moduleRegistry.get(this._testPath || ''),
```

Comment:
```suggestion
      value: this._testPath ? this._moduleRegistry.get(this._testPath) : null,
```

Probably the same in practice, but more explicit